src/DataSource.ts

```diff
@@ -1,158 +1,56 @@
-import { defaults, zip } from 'lodash';
+import { DataQueryRequest, DataQueryResponse, DataSourceApi, DataSourceInstanceSettings } from '@grafana/data';
+import { FetchResponse } from '@grafana/runtime';
 
-import {
-  DataQueryRequest,
-  DataQueryResponse,
-  DataSourceApi,
-  DataSourceInstanceSettings,
-  FieldType,
-  MutableDataFrame,
-} from '@grafana/data';
-import { BackendSrvRequest, FetchResponse, getBackendSrv } from '@grafana/runtime';
+import { CmkQuery, DataSourceOptions, Edition, Backend as BackendType } from './types';
+import { buildRequestBody, WebApiResponse } from './webapi';
 
-import { CmkQuery, DataSourceOptions, defaultQuery, Edition } from './types';
-import {
-  buildRequestBody,
-  buildUrlWithParams,
-  createWebApiRequestBody,
-  createWebApiRequestSpecification,
-  WebAPiGetGraphResult,
-  WebApiResponse,
-} from './webapi';
+import WebApiBackend from './backend/web';
+import RestApiBackend from './backend/rest';
+import { Backend } from './backend/types';
 
 export class DataSource extends DataSourceApi<CmkQuery> {
+  webBackend: WebApiBackend;
+  restBackend: RestApiBackend;
+
   constructor(private instanceSettings: DataSourceInstanceSettings<DataSourceOptions>) {
     super(instanceSettings);
+    this.webBackend = new WebApiBackend(this);
```

Comment:
no, because we need the webbackend for autocompletion. rest backend can not autocomplete.

Reply:
I'm not sure if that is conceptually the right place for the autocompleter stuff, but it would be a bigger refactoring with no value but aesthetics. So let's just leave it there.